qa/common/test_util.py

```diff
@@ -230,3 +232,40 @@ def get_dyna_sequence_model_name(pf, dtype):
 
 def get_zero_model_name(pf, io_cnt, dtype):
     return "{}_zero_{}_{}".format(pf, io_cnt, np.dtype(dtype).name)
+
+
+class InferUnit(unittest.TestCase):
+    # InferUnit stores test result and prints it to stdout. In order to use
+    # this class, unittest class should inherit this class
+
+    current_result = None
+
+    @classmethod
+    def setResult(cls, amount, errors, failures, skipped):
+        cls.amount, cls.errors, cls.failures, cls.skipped = \
+            amount, errors, failures, skipped
+
+    def tearDown(self):
+        # this is called immediately after the test method
```

Comment:
So current_result contains the aggregate result not only of the test that just finished but also of all previously run tests? Otherwise I don't see where you are accumulating the counts.

Reply:
`current_result` contains the aggregate count after all the test methods finish execution. We only need to pass the same result object to `run` so that it accumulates the results. As far as I have checked, each `run` is called once for every test method in the UnitTest class (e.g. InferTest).